scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -144,9 +178,13 @@
     ; Initialize the pod watch path.
     (let [conn cook.datomic/conn
           cook-pod-callback (make-cook-pod-watch-callback this)]
+      ; We set expected state first because initialize-pod-watch sets (and invokes callbacks on and reacts to) the expected and the gruadually discovere existing.
+      (reset! expected-state-map (determine-expected-state-on-startup conn api-client name running-task-ents))
+
       (api/initialize-pod-watch api-client all-pods-atom cook-pods-atom cook-pod-callback)
-      ; Before we execute determine-expected-state-on-startup, we need to ensure that initialize-pod-watch sets cook-pods-atom.
-      (reset! expected-state-map (determine-expected-state-on-startup conn name running-task-ents cook-pods-atom)))
+      (if scan-frequency-seconds-config
```

Comment:
Since this has a default value, can it actually be disabled? Should we have some other way to disable it if it's defaulted to every 2 minutes?

Reply:
You're not correct if you don't scan, because without the scan, you'll *never* retry any transient problems. (datomic or otherwise.) That said, just set it to 60 million seconds and it'll only retry every 2 years.

Reply:
I just mean, can we delete the if? Since this will never be non-nil?

Reply:
I expect to set it to null in unit tests.

Reply:
Ah, makes sense.